zkevm-circuits/src/sig_circuit.rs

```diff
@@ -245,105 +188,105 @@ impl<F: Field> SignVerifyConfig<F> {
+/// key corresponding to an Ethereum Address.
+#[derive(Clone, Debug, Default)]
+pub struct SigCircuit<F: Field> {
+    /// Max number of verifications
+    pub max_verif: usize,
+    /// Without padding
+    pub signatures: Vec<SignData>,
+    /// Marker
+    pub _marker: PhantomData<F>,
 }
 
-/// Temp struct to hold the intermediate data; removing life timer.
-// Issue with life timer:
-//
-// Suppose we have two piece of codes, that request different regions/contexts from the layouter.
-// The first piece of the code will return an `assigned_cell` that is to be used by the second code
-// piece. With halo2 we can safely pass this `assigned_cell` around. They are bounded by a life
-// timer `'v` which is when the field element is created.
-//
-// Now in halo2-lib, there is an additional life timer which says an `assigned_cell` cannot outlive
-// the `region` for which this cell is created. (is this understanding correct?)
-// That means the output cells of the first region cannot be passed to the second region.
-//
-// To temporary resolve this issue, we create a temp struct without life timer.
-// This works with halo2-lib/pse but not halo2-lib/axiom.
-// We do not support halo2-lib/axiom.
-//
-// NOTE: this is a temp issue with halo2-lib v0.2.2.
-// with halo2-lib v0.3.0 the timers are already removed.
-// So we don't need this temp fix once we sync with halo2-lib audited version.
-#[derive(Debug, Clone)]
-pub(crate) struct AssignedValueNoTimer<F: Field> {
-    pub cell: Cell,
-    pub value: Value<F>,
-    pub row_offset: usize,
-    pub context_id: usize,
-}
+impl<F: Field> SubCircuit<F> for SigCircuit<F> {
+    type Config = SigCircuitConfig<F>;
 
-impl<'v, F: Field> From<AssignedValue<'v, F>> for AssignedValueNoTimer<F> {
-    fn from(input: AssignedValue<'v, F>) -> Self {
-        Self {
-            cell: input.cell(),
-            value: input.value,
-            row_offset: input.row_offset,
-            context_id: input.context_id,
+    fn new_from_block(block: &crate::witness::Block<F>) -> Self {
+        SigCircuit {
+            // TODO: seperate max_verif with max_txs?
+            max_verif: block.circuits_params.max_txs,
+            // TODO: better way than unwrap?
+            signatures: block.get_sign_data(true).unwrap(),
+            _marker: Default::default(),
         }
     }
-}
 
-impl<'v, F: Field> From<AssignedValueNoTimer<F>> for AssignedValue<'v, F> {
-    fn from(input: AssignedValueNoTimer<F>) -> Self {
-        Self {
-            cell: input.cell,
-            value: input.value,
-            row_offset: input.row_offset,
-            _marker: PhantomData::default(),
-            context_id: input.context_id,
-        }
+    fn synthesize_sub(
+        &self,
+        config: &Self::Config,
+        challenges: &Challenges<Value<F>>,
+        layouter: &mut impl Layouter<F>,
+    ) -> Result<(), Error> {
+        config.ecdsa_config.range.load_lookup_table(layouter)?;
```

Comment:
This is already loaded during config. do we need to load it again?

Reply:
<del>oh i should do this in assignment. not in config.</del>

no i think it is not loaded during config?